src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -717,11 +717,12 @@ public ResponseEntity<StudentExam> submitStudentExam(@PathVariable Long courseId
         if (studentExam.isSubmitted()) {
             throw new BadRequestException();
         }
-        if (studentExam.getIndividualEndDateWithGracePeriod().isAfter(now())) {
-            throw new AccessForbiddenException("Exam", examId);
+        ZonedDateTime submissionTime = now();
+        if (studentExam.getIndividualEndDateWithGracePeriod().isAfter(submissionTime)) {
+            throw new AccessForbiddenException("FORBIDDEN: You tried to toggle a student exam " + studentExamId + " to submitted before the individual end date "
```

Comment:
This message is currently never actually displayed on the client. If this case happens, 2 popups will appear:
<img width="817" alt="grafik" src="https://github.com/ls1intum/Artemis/assets/18218285/88aafe2c-df23-49e2-8bf2-c5bdc3cd5dac">
The first one is a manual error handling popup, the second one is the popup created from this exception.

I think it would make sense to change this to an `AccessForbiddenAlertException` and provide translations in the client (and ideally remove the manual popup). That way the user sees the actual error that happened, instead of two generic errors.

Reply:
I agree, I was asking someone to improve this in a follow-up

Reply:
@florian-glombik

Reply:
The issue is addressed in https://github.com/ls1intum/Artemis/pull/7337